python/ray/serve/_private/deployment_state.py

```diff
@@ -2194,6 +2194,23 @@ def update(self):
             del self._deployment_states[tag]
 
         if len(deleted_tags):
+            self._record_deployment_usage()
             record_extra_usage_tag(
                 TagKey.SERVE_NUM_DEPLOYMENTS, str(len(self._deployment_states))
             )
+
+    def _record_deployment_usage(self):
```

Comment:
It is a little bit weird we only get the stats when the deployment is deleted...(I'd prefer to to do it on the client side directly), WDYT?

Reply:
We also get it when the deployment is added– I just forgot to modify that piece of code. I've updated it.

My overall concern about tracking these metrics in the client is that a Ray cluster has to be running for these metrics to get tracked. When we run the metrics in actors, such as the Serve controller, it's guaranteed that a Ray cluster is running.